RUN-4298 - Remove channels from middleware

tgoc99 opened the pull request:
Removed channels logic from the middleware and made it follow the normal protocol.  Return undefined so that it does not automatically ack back, then later use ack function.  

Relies on this [js-adapter PR](https://github.com/HadoukenIO/js-adapter/pull/155)
Builds on this [core PR](https://github.com/HadoukenIO/core/pull/494)

Still passes these channels v2API tests:
[Test1](https://testing-dashboard.openfin.co/#/app/tests/5b560fd854b21953031f37a6/edit)
[Test2](https://testing-dashboard.openfin.co/#/app/tests/5b560f8f54b21953031f37a4/edit)
[Test3](https://testing-dashboard.openfin.co/#/app/tests/5b560f5454b21953031f37a3/edit)
[Test4](https://testing-dashboard.openfin.co/#/app/tests/5b560f0e54b21953031f37a2/edit)
[Test5](https://testing-dashboard.openfin.co/#/app/tests/5b56093354b21953031f379d/edit)
[Test6](https://testing-dashboard.openfin.co/#/app/tests/5b55ff0d54b21953031f3799/edit)
[Test7](https://testing-dashboard.openfin.co/#/app/tests/5b5608ce54b21953031f379c/edit)
[Test8](https://testing-dashboard.openfin.co/#/app/tests/5b56079e54b21953031f379b/edit)
[Test9](https://testing-dashboard.openfin.co/#/app/tests/5b56071654b21953031f379a/edit)

Tests: 
[Win7](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5b59ef1054b21953031f3811)
[Win10](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5b59f07554b21953031f3812)

pbaize commented on src/browser/api/channel.ts:
```diff
@@ -22,7 +51,11 @@ export module Channel {
     }
 
     // Could be any identifier
+<<<<<<< HEAD
```
Merge conflict marker

pbaize commented on src/browser/api/channel.ts:
```diff
@@ -61,4 +94,100 @@ export module Channel {
 
         return providerIdentity;
     }
+<<<<<<< HEAD
```
Here too

tgoc99 commented:
updated testrunners:

[Win7](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5b5b3c3154b21953031f3827)
[Win10](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5b5b3d7f54b21953031f3828)

One Channel test failed bc I updated it for next PR (getallChannels). Re-ran with that commented out and it passed.
